src/SFML/Network/TcpSocket.cpp

```diff
@@ -245,12 +245,15 @@ Socket::Status TcpSocket::send(const void* data, std::size_t size, std::size_t&
     int result = 0;
     for (sent = 0; sent < size; sent += static_cast<std::size_t>(result))
     {
+#if !(defined(_MSC_VER) && defined(__clang__))
         #pragma GCC diagnostic push
         #pragma GCC diagnostic ignored "-Wuseless-cast"
+#endif
```

Comment:
We should decide if we keep preprocessor directives indented at the beginning of the line, or make it dependent on scope -- but not mix the two styles.

Reply:
I'm inclined to put all preprocessor code on the left margin since the preprocessor has no concept of scope nor do scopes have any concept of the preprocessor. I'd be willing to make a PR unifying preprocessor style since the library is already using a mix of styles.

Reply:
Given what @ChrisThrasher said with these likely to go in on a future commit then I can make sure they all follow a consistent indentation level wherever I re-add them. I'll revert them from this commit for now.